### PR TITLE
Update SimPEG.Report()

### PR DESCRIPTION
**Pull request by prisae**
- Update the optional package list to include all optional imports.
- Move `requires` ahead of `memProfileWrapper`, to decorate latter with `@requires('memory_profiler')`.

### CODE REVIEW

**Comment by prisae** [comment]
We could add the optional imports `omf`, `pyvista`, and `mpl_toolkits` from `discretize`. Opinions?

File: SimPEG/utils/code_utils.py [hunk 1]
```diff
@@ -325,16 +326,32 @@ def __init__(self, add_pckg=None, ncol=3, text_width=80, sort=False):
         core = [
             "SimPEG",
             "discretize",
-            "pymatsolver",
+            "pymatsolver",  # TODO: Will be replaced by pydiso
```
Comment (jcapriot) [inline SimPEG/utils/code_utils.py:329]: It won't be replaced by pydiso. pydiso is specifically a wrapper for intel's pardiso libraries. pymatsolver is another level above that provide abstract operations with solver objects.

File: SimPEG/utils/code_utils.py [hunk 2]
```diff
@@ -325,16 +326,32 @@ def __init__(self, add_pckg=None, ncol=3, text_width=80, sort=False):
...
             "vectormath",
-            "properties",
+            "properties",   # TODO: Will be removed eventually
```
Comment (jcapriot) [inline SimPEG/utils/code_utils.py:331]: properties (and vectormath) are no longer required by SimPEG.

File: SimPEG/utils/code_utils.py (hunk 1 above)
Comment (prisae) [reply SimPEG/utils/code_utils.py:471]: Oh, thanks for the clarification. Should pydiso be another optional or mandatory package then?

**Comment by jcapriot** [comment]
> Oh, thanks for the clarification. Should pydiso be another optional or mandatory package then?

This does beg a question:
Should the reported requirements also include requirements of the required packages?

**Comment by prisae** [comment]
> Should the reported requirements also include requirements of the required packages?

Good question. Not sure, something to be discussed. I am not sure if the list I put together is any good. I think it should contain those packages that are most important for issues/failures/changes/dependencies/whatever-is-needed-in-debugging.

**Comment by prisae** [comment]
Simply the most important ones. Because anything else could be asked for specifically, e.g.,
```
SimPEG.Report(['vtk', 'ipywidgets'])
```
if needed.

**Pull request merged**